internal/manifests/compactor.go

```diff
@@ -106,26 +107,27 @@ func NewCompactorStatefulSet(opt Options) *appsv1.StatefulSet {
 		},
 		ObjectMeta: metav1.ObjectMeta{
 			Name:   fmt.Sprintf("loki-compactor-%s", opt.Name),
-			Labels: compactorLabels,
+			Labels: l,
 		},
 		Spec: appsv1.StatefulSetSpec{
 			PodManagementPolicy:  appsv1.OrderedReadyPodManagement,
 			RevisionHistoryLimit: pointer.Int32Ptr(10),
 			Replicas:             pointer.Int32Ptr(opt.Stack.Template.Compactor.Replicas),
 			Selector: &metav1.LabelSelector{
-				MatchLabels: labels.Merge(compactorLabels, GossipLabels()),
+				MatchLabels: labels.Merge(l, GossipLabels()),
 			},
 			Template: corev1.PodTemplateSpec{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:   fmt.Sprintf("loki-compactor-%s", opt.Name),
-					Labels: labels.Merge(compactorLabels, GossipLabels()),
+					Name:        fmt.Sprintf("loki-compactor-%s", opt.Name),
+					Labels:      labels.Merge(l, GossipLabels()),
+					Annotations: a,
```

Comment:
shouldn't this be higher up at the statefulset meta level, not the pod template?

Reply:
You need this annotation to restart the pods only. Adding them like [kubectl](https://github.com/kubernetes/kubectl/blob/release-1.20/pkg/polymorphichelpers/objectrestarter.go#L32) on the pod template doesn't bother the k8s deployment controller just the scheduler.